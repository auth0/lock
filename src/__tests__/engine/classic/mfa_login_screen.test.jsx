import React from 'react';
import { mount } from 'enzyme';

import { expectComponent, extractPropsFromWrapper, mockComponent } from 'testUtils';

jest.mock('connection/database/mfa_pane', () => mockComponent('mfa_pane'));

//there's a circular dependency with this module, so we need to mock it
jest.mock('engine/classic');

const getComponent = () => {
  const MFALoginScreen = require('engine/classic/mfa_login_screen').default;
  const screen = new MFALoginScreen();
  return screen.render();
};

describe('MFALoginScreen', () => {
  beforeEach(() => {
    jest.resetModules();

    jest.mock('connection/database/index', () => ({
      hasScreen: () => false
    }));

    jest.mock('connection/database/actions', () => ({
      cancelMFALogin: jest.fn(),
      logIn: jest.fn()
    }));

    jest.mock('core/signed_in_confirmation', () => ({
      renderSignedInConfirmation: jest.fn()
    }));
  });
  const defaultProps = {
    i18n: {
      str: (...keys) => keys.join(',')
    },
    model: 'model'
  };
  it('renders correctly', () => {
    const Component = getComponent();

    expectComponent(<Component {...defaultProps} />).toMatchSnapshot();
  });
});
