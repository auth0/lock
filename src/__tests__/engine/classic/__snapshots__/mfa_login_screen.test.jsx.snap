// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MFALoginScreen renders correctly 1`] = `
<div
  data-__type="mfa_pane"
  data-i18n={
    Object {
      "str": [Function],
    }
  }
  data-instructions="mfaLoginInstructions"
  data-lock="model"
  data-mfaInputPlaceholder="mfaInputPlaceholder"
  data-title="mfaLoginTitle"
/>
`;
